### DISCUSSION TITLE
How do I specify the peer dependency version of an npm package?

### DISCUSSION THREAD

**Discussion by TomokiMiyauci** [denoland/deno#22460] [Q&A]
If I use npm specifier and import a package that contains a peer dependency, I want to specify the version of that peer dependency.

**Comment by nnmrts** [comment]
What is the specific problem you are encountering? Usually packages specify their peer dependencies on their own and if you want to use a different version of that depended on package directly, you can just import or add it to your import map or deps.ts on your own as well, which of course might lead to incompatibilities.

**Reply by TomokiMiyauci** [reply]
For example, if I use `npm:react@canary` in my project, I want all packages that have `react` as a peer dependency to use it.

I thought this could be solved with `scopes` in importmaps, but did not know how it should be specified.

**Comment by TomokiMiyauci** [comment]
Closed because the issue was already there.
#20868
